src/TimeIntegrator.C

```diff
@@ -178,7 +178,12 @@ void TimeIntegrator::prepare_for_time_integration()
 
   // possible restart; need to extract current time (max wins)
   for ( ii = realmVec_.begin(); ii!=realmVec_.end(); ++ii) {
+    const double time_coming_in = currentTime_;
     currentTime_ = std::max(currentTime_, (*ii)->populate_restart(timeStepNm1_, timeStepCount_));
+    // we can do exact comparison because the numbers will be bit-wise exact the
+    // same unless the value was changed by the restart
+    if (time_coming_in != currentTime_)
+      overset_->update_connectivity();
```

Comment:
@psakievich Why is this needed following - https://github.com/Exawind/nalu-wind/pull/924/files#diff-fecc3b6f108bff965d05cbaa485de2d60d273e8f71457705898ac4dd7909c36dR3216 ? What would be the use case here ?

Reply:
@ashesh2512 `prepare_for_time_integration` is called every timestep but we only need to do this extra update if there was a restart. https://github.com/Exawind/nalu-wind/blob/ad5ffdf22df0f5ef174383759359978d52de61f3/src/TimeIntegrator.C#L307-L312

Reply:
Moving this call outside the realm vector loop and removing the call inside realm.